interfaces/connection.go

```diff
@@ -49,29 +50,46 @@ type ConnectedSlot struct {
 // Attrer is an interface with Attr getter method common
 // to ConnectedSlot, ConnectedPlug, PlugInfo and SlotInfo types.
 type Attrer interface {
-	Attr(key string, val interface{}) error
+	Attr(path string, val interface{}) error
+	Lookup(path string) (interface{}, bool)
 }
 
-func getAttribute(snapName string, ifaceName string, staticAttrs map[string]interface{}, dynamicAttrs map[string]interface{}, key string, val interface{}) error {
+func lookupAttr(staticAttrs map[string]interface{}, dynamicAttrs map[string]interface{}, path string) (interface{}, bool) {
 	var v interface{}
-	var ok bool
+	comps := strings.Split(path, ".")
```

Comment:
What happens on consecutive dots? I don't think this is what you wanted. How about `strings.FieldsFunc(path, func(r rune) bool { return r == '.' })`.

Reply:
Ok, this wouldn't misbehave as it would just fail on finding attributes, but I agree, it's better to be  explicit, so I changed it per your suggestion, thanks!